src/shapes/IText/__snapshots__/IText.test.ts.snap

```diff
@@ -0,0 +1,56 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`IText cursor width under a group scaled by 1 and canvas zoomed by 1 1`] = `
+[
+  [
+    -92.2,
```

Comment:
Aren't you a fan of Playwright? At least you could catch a visual regression. These kind of tests instead easily break if the value changes only a little. A better test, in Jest, would be checking that cursors at different group scale have the same size, whatever is the value.

Reply:
I agree snapshots tests shouldn't be used so largely.

Reply:
Well guys I first tried the playwright test and it won't be percise enough to catch regression I belive.
I can add it, it is stashed on my local branch and try to refine it... but I don't see how.



> I agree snapshots tests shouldn't be used so largely.

I prefer snapshots instead of hardcoding because then it is easy to update and see the diff. The main pro of jest IMO.

Reply:
> I prefer snapshots instead of hardcoding because then it is easy to update and see the diff. The main pro of jest IMO.

I suggested something like `expect(getCursoWidth(renderWithScale1)).toEqual(getCursoWidth(renderWithScale2))`, then you don't need to hardcode anything or even care about what is the value

> it is easy to update and see the diff. The main pro of jest IMO.

Jest snapshots are definitely a nice feature, but as everything you have to use it properly. They have drawbacks:
1. Artifacts like the snapshot files pollute the codebase
2. People tend to **too easily** just update a diff thinking it's okay because you can just press a keystroke `u` and voilà, the test passes. A normal test forces you to look into the problem and really assert whether it was a false negative.

I've been using Jest since React 0.14, trust me a bit 😄

Reply:
yes my concern is indeed the easyness of update.
In this case i slimmed down what we are snasphotting so that it makes clearer what are we update.
In the case like before with an array of 4 numbers, if we change the way we draw cursor, like we just center it in place, the first 2 numbers of the snapshot would have changed, but with no indication that those are the values of x,y of fillRect.

In generale is preferable to have tests that says:
expect(mockCalls[0][2], 'width is exactly 100').toBe(text.cursorWidth);

or something in that ballpark